src/OrchardCore/OrchardCore/Modules/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -520,7 +520,7 @@ private static void AddAuthentication(OrchardCoreBuilder builder)
             .Configure(app =>
             {
                 app.UseAuthentication();
-            });
+            }, order: -150);
```

Comment:
@Piedone changing authentication order fixes the issue, so no need to touch the order of the localization module anymore

Reply:
This looks like it ultimately has the same issue as changing the order in the `Startup` class, i.e. https://github.com/OrchardCMS/OrchardCore/pull/13181#discussion_r1091643831 I don't think we can properly assess what this might affect, since it reorders the pipeline for everyone, having potentially unwanted effects that are hard to debug.

Reply:
> This looks like it ultimately has the same issue as changing the order in the Startup class, i.e. https://github.com/OrchardCMS/OrchardCore/pull/13181#discussion_r1091643831 I don't think we can properly assess what this might affect, since it reorders the pipeline for everyone, having potentially unwanted effects that are hard to debug.

I don't see any issue reordering the auth middleware to be registered before anything else. The opposite would have been an issue.

Reply:
I'm really not sure.

Anyway, are you sure, Hisham, that both this and the Manifest `Priority` are needed? If yes, document the reason for doing this inline here too, and add a note to the release notes.

Reply:
Both are necessary to provide a consistent result.

Reply:
Also, the fact that it is required to be adjusted to be registered before other services kind of relevates that it was a bug. The auth middleware needs to be registered really soon in the pipeline if we want to make use of it.

Reply:
I hope you're right.